packages/core/src/js/integrations/devicecontext.ts

```diff
@@ -84,9 +85,27 @@ async function processEvent(event: Event): Promise<Event> {
     ? native['breadcrumbs'].map(breadcrumbFromObject)
     : undefined;
   if (nativeBreadcrumbs) {
-    const maxBreadcrumbs = getClient()?.getOptions().maxBreadcrumbs ?? 100; // Default is 100.
-    event.breadcrumbs = nativeBreadcrumbs.concat(event.breadcrumbs || []).slice(0, maxBreadcrumbs);
+    // Concatenate nativeBreadcrumbs first, then event.breadcrumbs
+    event.breadcrumbs = nativeBreadcrumbs.concat(event.breadcrumbs || []);
   }
 
+  const options = getClient()?.getOptions();
+  const maxBreadcrumbs = options.maxBreadcrumbs ?? 100; // Default is 100.
+  const devServerUrl = getDevServer()?.url || '';
+  const dsn = options.dsn || '';
+
+  let allBreadcrumbs = event.breadcrumbs || [];
+
+  // Filter out Dev Server and Sentry DSN request breadcrumbs
+  allBreadcrumbs = allBreadcrumbs.filter((breadcrumb: Breadcrumb) => {
```

Comment:
This integration will be execute only for JS events, so native errors and replays would still contain the dev server network breadcrumbs.

Reply:
Could use the `beforeBreadcrumbs` callbacks in JS, Cocoa and Android.

Reply:
Thank you for the feedback and hint @krystofwoldrich 🙇